### ISSUE TITLE
Noto Devanagari: there is a bug

### ISSUE THREAD

**Issue by Abhay2Agarwal** [notofonts/devanagari#47]
I was recently typing in Devanagari, and i encountered a bug, which i think is significant. The description is :

र + ् + [any character] + ो + र + े
र्रोरे

reph overlaps with matra e

I will attach some photos

![20241005_125715.jpg](https://github.com/user-attachments/assets/b220b5b8-a2cd-4bb0-99af-028b1bfd83f1)

![20241005_125743.jpg](https://github.com/user-attachments/assets/f451142f-9559-437f-a249-e7daf402916e)

A lot of fonts have this bug

![Screenshot_20241005_125109_Chrome.jpg](https://github.com/user-attachments/assets/c29ffdf6-c7d5-4b82-8b67-b2a42b851710)

I hope you will see it and fix this.

**Comment by Abhay2Agarwal** [comment]
This is evident while writing कार्पोरेशन बैंक. You can see "र्पोरे" reph & matra e do overlap

**Comment by simoncozens** [comment]
I agree that it overlaps - but I am not sure that it's a bug! Is it unreadable? Can it be confused for something else? As you say, it's something that happens with most Devanagari fonts - so maybe it's OK; maybe it's expected at this point.

I could add an alternate, steeper e-matra just for this case, but it's adding another glyph and rules to fix a rare corner case, so I'd like to make sure it's actually a problem first.

**Comment by Abhay2Agarwal** [comment]
> I agree that it overlaps - but I am not sure that it's a bug! Is it unreadable? Can it be confused for something else? As you say, it's something that happens with most Devanagari fonts - so maybe it's OK; maybe it's expected at this point.
> 
> I could add an alternate, steeper e-matra just for this case, but it's adding another glyph and rules to fix a rare corner case, so I'd like to make sure it's actually a problem first.


Yes in smaller font sizes it is creating confusion and unreadable. Please fix this. 

Maybe another way to get this better is modify either /both reph & matra e so they never lap.

I was recently writing a mail to the corporation bank (an indian bank) in Hindi and i myself got confused looking at the way we wrote कार्पोरेशन.

**Comment by Abhay2Agarwal** [comment]
This overlap will also happen on:

र्खरे र्गरे र्घरे ... र्शरे र्षरे र्सरे

Most of the alphabets in hindi.

I would gently urge you to modify reph a little :)

**Comment by MAZ06** [comment]
Related:

र्परै
<img width="353" height="525" alt="Image" src="https://github.com/user-attachments/assets/263eff03-3faa-4302-bd12-1567a96cc737" />

**Comment by Abhay2Agarwal** [comment]
> Worth mentioning is that this does *not* happen in Noto Serif Devanagari:
> ![Image](https://github.com/user-attachments/assets/9b9b7e22-6a96-4423-b173-126f450022a8)
> 
> It seems that the *reason* why it doesn't happen is because Noto Serif Devanagari has two different glyphs probably for this exact situation:
> ![Image](https://github.com/user-attachments/assets/bbba44b9-9650-4767-8d73-a17f14d97972)
> 
> I see no reason why Noto Sans Devanagari shouldn't have this feature as well. In fact, *why* doesn't Noto Sans Devanagari have it already?

It does overlap in Noto Serif Devanaagari too in condensed and small font size.
